src/deluge/hid/display/oled.cpp

```diff
@@ -434,89 +434,187 @@ void OLED::sendMainImage() {
 struct TextLineBreakdown {
 	char const* lines[TEXT_MAX_NUM_LINES];
 	uint8_t lineLengths[TEXT_MAX_NUM_LINES];
+	uint32_t lineWidths[TEXT_MAX_NUM_LINES];
 	int32_t numLines;
 	int32_t longestLineLength;
 	int32_t maxCharsPerLine;
+	int32_t longestLineWidth;
+	int32_t maxWidthPerLine;
 };
 
-void breakStringIntoLines(char const* text, TextLineBreakdown* textLineBreakdown) {
+/// this function takes a string and breaks it into multiple lines
+/// it does so by iterating through each character in a string, calculating the width in pixels
+/// for each character and comparing that width to the maximum width of a line
+/// as characters are processed, the width and length of a given line is incremented so that it can be
+/// compared against. If a character cannot be drawn because a line's width and length is reached
+/// then that character is evaluated to determine what to do with it
+/// for example:
+/// a) if the character is part of a word, then that word will be rendered on the next line
+/// b) if the character is a space, then the space gets ignored and the next character will be drawn on
+///    the next line
+void breakStringIntoLines(char const* text, TextLineBreakdown* textLineBreakdown, int32_t textHeight) {
```

Comment:
> can this be broken into more functions?
> 
> E.g. find length of word, find length of string, find string split points (e.g. an array of the string length after each word)

When you get a chance can you look at what I did, I don't think I split it up exactly as you wanted (e.g. no array).

But I'm not entirely sure on how to do the array thing since the number of elements of the array could vary based on number of words fit into the line.

Reply:
yeah it would have to return a vector, it's not really important though. Just how I'd have done it but this looks fine!